File: 1817-calculate-money-in-leetcode-bank/1817-calculate-money-in-leetcode-bank.py
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks = n // 7
        days = n % 7

        total = weeks * 28 + (weeks * (weeks - 1) // 2) * 7
        for i in range(days):
            total += (weeks + 1) + i
        return total

File: 1817-calculate-money-in-leetcode-bank/1817-calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int weeks = n / 7;
        int days = n % 7;

        int total = weeks * 28 + (weeks * (weeks - 1) / 2) * 7;
        for (int i = 0; i < days; i++) {
            total += (weeks + 1) + i;
        }
        return total;
    }
}
